google-cloud-bigquery/lib/google/cloud/bigquery/job.rb

```diff
@@ -239,12 +244,16 @@ def reload!
         #                               format: "json"
         #   extract_job.wait_until_done!
         #   extract_job.done? #=> true
-        def wait_until_done!
-          backoff = ->(retries) { sleep 2 * retries + 5 }
+        def wait_until_done! min: 5, max: Float::INFINITY
```

Comment:
I _think_ I would like to see the method signature be the following:

```ruby
def wait_until_done! initial_delay: 5, delay_multiplier: 2, max_delay: 300
```

The current implementation could be considered to be the following:

```ruby
def wait_until_done! initial_delay: 5, delay_multiplier: 2, max_delay: Float::INFINITY
```

But I think it would be better to cap the max delay at 5 minutes. @quartzmo, what do you think?